src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking.cs

```diff
@@ -17324,5 +17543,43 @@ static void F(A<object>? x, A<object?>? y)
                 //         w?.F.ToString();
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, ".F").WithLocation(17, 11));
         }
+
+        [Fact]
+        public void GroupBy()
+        {
+            var source =
+@"using System.Linq;
+class Program
+{
+    static void Main()
+    {
+        var items = from i in Enumerable.Range(0, 3) group (long)i by i;
+    }
+}";
+            var comp = CreateCompilationWithMscorlibAndSystemCore(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void ExplicitTypeArguments()
+        {
+            var source =
+@"interface I<T> { }
+class C
+{
+    C P => throw new System.Exception();
+    I<T> F<T>(T t)
+    {
+        throw new System.Exception();
+    }
+    static void M(C c)
+    {
+        c.P.F<object>(string.Empty);
+        (new[]{ c })[0].F<object>(string.Empty);
+    }
+}";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics();
```

Comment:
>comp.VerifyDiagnostics(); [](start = 12, length = 25)

Do we rely on the fact that mscorlib is not annotated here?

Reply:
Added comment that this test is for `NullableWalker.HasImplicitTypeArguments` rather than for nullable analysis.

---
In reply to: [169861930](https://github.com/dotnet/roslyn/pull/24935#discussion_r169861930) [](ancestors = 169861930)